tensorrt_llm/serve/tool_parser/deepseekv32_parser.py

```
@@ -172,21 +172,31 @@ def parse_streaming_increment(self, new_text: str, tools: List[Tool]) -> Streami
-        # Check if buffer contains any DSML markers or ends with potential tag prefix
-        # This handles partial/streaming DSML content
-        dsml_markers = ["｜DSML｜", "<｜", "</｜"]
-        potentially_dsml = any(marker in current_text for marker in dsml_markers)
-
-        # Also check if text ends with start of a tag (to handle "<" arriving separately)
-        dsml_prefixes = ["<", "<｜", "</", "</｜"]
-        ends_with_prefix = any(current_text.rstrip().endswith(prefix) for prefix in dsml_prefixes)
+        # Hold the buffer back only while its tail could still complete into one of
+        # the DSML delimiters. Testing whether a marker appears anywhere instead
+        # would keep the buffer forever once ordinary text diverges from a delimiter,
+        # for example "<｜DSML｜function" followed by "ality".
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `｜` (FULLWIDTH VERTICAL LINE). Did you mean `|` (VERTICAL LINE)?

(RUF003)